src/promptflow/promptflow/_sdk/operations/_local_azure_connection_operations.py

```diff
@@ -16,20 +17,38 @@
 
 class LocalAzureConnectionOperations(TelemetryMixin):
     def __init__(self, connection_provider, **kwargs):
-        from azure.identity import DefaultAzureCredential
 
         from promptflow.azure._pf_client import PFClient as PFAzureClient
 
         super().__init__(**kwargs)
         subscription_id, resource_group, workspace_name = self._extract_workspace(connection_provider)
         self._pfazure_client = PFAzureClient(
             # TODO: disable interactive credential when starting as a service
-            credential=DefaultAzureCredential(exclude_interactive_browser_credential=False),
+            credential=self.get_credential(),
             subscription_id=subscription_id,
             resource_group_name=resource_group,
             workspace_name=workspace_name,
         )
 
+    @classmethod
+    def get_credential(cls):
+        from azure.identity import DefaultAzureCredential, DeviceCodeCredential
+
+        if is_from_cli():
+            try:
+                # Try getting token for cli without interactive login
+                credential = DefaultAzureCredential()
+                credential.get_token("https://management.azure.com/.default")
+            except Exception:
+                print_red_error("Please run 'az login' to set up account.")
```

Comment:
let's adopt the hint message shared by AI CLI guys

Reply:
We should display 
default: az login 
codespace: az login --use-device-identiy
CI: az login --identy  # let's sync on this